### PR TITLE
Added PostgreSQL enablement to documentation

### PR DESCRIPTION
**Pull request by russjones**

### CODE REVIEW

**Review by r0mant — APPROVED** [review]

File: docs/pages/reference/backends.mdx [hunk 1]
```diff
@@ -350,6 +370,134 @@ credentials, or [managed
 identity](https://learn.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview)
 credentials.
 
+### Development
+
+If you are not ready to connect Teleport to a production instance of
+PostgreSQL, you can use the following instructions to setup a throwaway
```
Comment (ptgott) [inline docs/pages/reference/backends.mdx:376]:
```suggestion
PostgreSQL, you can use the following instructions to set up a throwaway
```

File: docs/pages/reference/backends.mdx [hunk 2]
```diff
@@ -350,6 +370,134 @@ credentials, or [managed
...
+instance of PostgreSQL using Docker.
+
+First copy the following script to disk and run it to generate the CA, client
+certificate, and server certificate used by Teleport and PostgreSQL to
+establish a secure mutually authenticated connection.
```
Comment (ptgott) [inline docs/pages/reference/backends.mdx:381]:
```suggestion
establish a secure mutually authenticated connection:
```

File: docs/pages/reference/backends.mdx [hunk 3]
```diff
@@ -350,6 +370,134 @@ credentials, or [managed
...
+create_certificate "client" "teleport"
+
+# Create server certificate with SAN.
+create_certificate "server" "localhost"
+
+echo "Certificates and keys generated successfully."
+```
+
+Next, create a `Dockerfile` using the [official PostgreSQL Docker
+image](https://hub.docker.com/_/postgres) and add `wal2json` to it.
```
Comment (ptgott) [inline docs/pages/reference/backends.mdx:427]:
```suggestion
image](https://hub.docker.com/_/postgres) and add `wal2json` to it:
```

File: docs/pages/reference/backends.mdx [hunk 4]
```diff
@@ -350,6 +370,134 @@ credentials, or [managed
...
+
+```
+FROM postgres:15.0
+RUN apt-get update
+RUN apt-get install -y postgresql-15-wal2json
+```
+
+Create a `init.sql` file that will ensure the Teleport user is created upon
+startup of the container.
```
Comment (ptgott) [inline docs/pages/reference/backends.mdx:436]:
```suggestion
startup of the container:
```

File: docs/pages/reference/backends.mdx [hunk 5]
```diff
@@ -350,6 +370,134 @@ credentials, or [managed
...
+Next, create a `Dockerfile` using the [official PostgreSQL Docker
+image](https://hub.docker.com/_/postgres) and add `wal2json` to it.
+
+```
+FROM postgres:15.0
+RUN apt-get update
+RUN apt-get install -y postgresql-15-wal2json
+```
+
+Create a `init.sql` file that will ensure the Teleport user is created upon
```
Comment (ptgott) [inline docs/pages/reference/backends.mdx:435]:
```suggestion
Create an `init.sql` file that will ensure the Teleport user is created upon
```

File: docs/pages/reference/backends.mdx [hunk 6]
```diff
@@ -350,6 +370,134 @@ credentials, or [managed
...
+startup of the container.
+
+```
+CREATE USER teleport WITH REPLICATION CREATEDB;
+```
+
+Create a `pg_hba.conf` file to enforce certificate based authentication for
+connections to PostgreSQL.
```
Comment (ptgott) [inline docs/pages/reference/backends.mdx:443]:
```suggestion
Create a `pg_hba.conf` file to enforce certificate-based authentication for
connections to PostgreSQL:
```

File: docs/pages/reference/backends.mdx [hunk 7]
```diff
@@ -350,6 +370,134 @@ credentials, or [managed
...
+
+```
+# TYPE  DATABASE        USER            CIDR-ADDRESS            METHOD
+local   all             all                                     trust
+hostssl all             all             ::/0                    cert
+hostssl all             all             0.0.0.0/0               cert
+```
+
+Create a `postgresql.conf` file that configures the WAL level and certificates
+used for authentication.
```
Comment (ptgott) [inline docs/pages/reference/backends.mdx:453]:
```suggestion
used for authentication:
```

File: docs/pages/reference/backends.mdx [hunk 8]
```diff
@@ -350,6 +370,134 @@ credentials, or [managed
...
+wal_level=logical
+max_replication_slots=10
+
+ssl=on
+ssl_ca_file='/certs/ca.crt'
+ssl_cert_file='/certs/server.crt'
+ssl_key_file='/certs/server.key'
+```
+
+Start the PostgreSQL container with the following command.
```
Comment (ptgott) [inline docs/pages/reference/backends.mdx:472]:
```suggestion
Start the PostgreSQL container with the following command:
```

File: docs/pages/reference/backends.mdx [hunk 9]
```diff
@@ -350,6 +370,134 @@ credentials, or [managed
...
+    -v $(pwd)/init.sql:/docker-entrypoint-initdb.d/init.sql \
+    -p 5432:5432 \
+    $(docker build -q .) \
+    postgres \
+    -c hba_file=/etc/postgresql/pg_hba.conf \
+    -c config_file=/etc/postgresql/postgresql.conf
+```
+
+Lastly, update the storage section in `teleport.yaml` to use PostgreSQL and
+start Teleport.
```
Comment (ptgott) [inline docs/pages/reference/backends.mdx:492]:
```suggestion
start Teleport:
```

File: docs/pages/reference/backends.mdx [hunk 10]
```diff
@@ -350,6 +370,134 @@ credentials, or [managed
...
+
+```
```
Comment (ptgott) [inline docs/pages/reference/backends.mdx:494]:
```suggestion
```yaml
```

**Review by ptgott — APPROVED** [review]
Approved with minor style/grammar nits

**Comment by ptgott** [comment]
@russjones Should we add backport labels to this PR?

**Pull request merged**